lib/kafka/sasl_plain_authenticator.rb

```diff
@@ -0,0 +1,37 @@
+module Kafka
+  class SaslPlainAuthenticator
```

Comment:
I'd like to see a test that exercises this code, maybe against a fake SASL server implementation. Currently none of this code is run during testing.